packages/contracts/src/contracts/current/protocol/Exchange/MixinSignatureValidator.sol

```diff
@@ -92,8 +92,15 @@ contract MixinSignatureValidator is
             LENGTH_GREATER_THAN_0_REQUIRED
         );
 
+        // Ensure signature is supported
+        uint8 signatureTypeRaw = uint8(popLastByte(signature));
+        require(
+            signatureTypeRaw < uint8(SignatureType.NSignatureTypes),
```

Comment:
Chatted about this offline. Going to leave it in there for completeness as an _unless the world is on fire, this should never happen_ edge case. It can be beneficial when debugging extreme cases; and also communicates that we considered all scenarios, which would be less clear if the function ended on an `else if {...}`.